3rdparty/mbedtls/0001-Backport-2632-code-changes-to-2.16.patch

```diff
@@ -0,0 +1,505 @@
+         MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_len( p, start, len ) );
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( p, start, MBEDTLS_ASN1_UTC_TIME ) );
++        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( p, start,
++                                             MBEDTLS_ASN1_UTC_TIME ) );
+     }
+     else
+     {
+@@ -356,15 +374,17 @@ static int x509_write_time( unsigned char **p, unsigned char *start,
+                                                   (const unsigned char *) t,
+                                                   size ) );
+         MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_len( p, start, len ) );
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( p, start, MBEDTLS_ASN1_GENERALIZED_TIME ) );
++        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( p, start,
++                                             MBEDTLS_ASN1_GENERALIZED_TIME ) );
+     }
+
+     return( (int) len );
+ }
+
+-int mbedtls_x509write_crt_der( mbedtls_x509write_cert *ctx, unsigned char *buf, size_t size,
+-                       int (*f_rng)(void *, unsigned char *, size_t),
+-                       void *p_rng )
++int mbedtls_x509write_crt_der( mbedtls_x509write_cert *ctx,
++                               unsigned char *buf, size_t size,
++                               int (*f_rng)(void *, unsigned char *, size_t),
++                               void *p_rng )
+ {
+     int ret;
+     const char *sig_oid;
+@@ -372,15 +392,14 @@ int mbedtls_x509write_crt_der( mbedtls_x509write_cert *ctx, unsigned char *buf,
+     unsigned char *c, *c2;
+     unsigned char hash[64];
+     unsigned char sig[SIGNATURE_MAX_SIZE];
+-    unsigned char tmp_buf[2048];
+     size_t sub_len = 0, pub_len = 0, sig_and_oid_len = 0, sig_len;
+     size_t len = 0;
+     mbedtls_pk_type_t pk_alg;
+
+     /*
+-     * Prepare data to be signed in tmp_buf
++     * Prepare data to be signed at the end of the target buffer
+      */
+-    c = tmp_buf + sizeof( tmp_buf );
++    c = buf + size;
+
+     /* Signature algorithm needed in TBS, and later for actual signature */
+
+@@ -406,27 +425,36 @@ int mbedtls_x509write_crt_der( mbedtls_x509write_cert *ctx, unsigned char *buf,
+     /* Only for v3 */
+     if( ctx->version == MBEDTLS_X509_CRT_VERSION_3 )
+     {
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_x509_write_extensions( &c, tmp_buf, ctx->extensions ) );
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_len( &c, tmp_buf, len ) );
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( &c, tmp_buf, MBEDTLS_ASN1_CONSTRUCTED |
+-                                                           MBEDTLS_ASN1_SEQUENCE ) );
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_len( &c, tmp_buf, len ) );
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( &c, tmp_buf, MBEDTLS_ASN1_CONTEXT_SPECIFIC |
+-                                                           MBEDTLS_ASN1_CONSTRUCTED | 3 ) );
++        MBEDTLS_ASN1_CHK_ADD( len,
```

Comment:
In this section for example, there does not seem to be any code difference; only format changes.

Reply:
Same observation as above, there are explicit formatting changes as part of the upstream PR. If you look closely, you'll see they don't match our clang-format rules.